gms/feature_service.cc

```diff
@@ -100,6 +104,14 @@ future<> feature_service::stop() {
     return make_ready_future<>();
 }
 
+#ifdef SCYLLA_ENABLE_ERROR_INJECTION
+void feature_service::get_suppressed_features() {
+    if (const auto features_list = utils::get_local_injector().inject_parameter<std::string_view>("suppress_features"); features_list) {
+        boost::split(_suppressed_features, *features_list, boost::is_any_of(";"));
+    }
```

Comment:
Not sure how it works. Isn't there a race between startup and the test controller? Or do we have something that ensures they will meet?

You can put the #ifdef in the function, so the caller and declaration can be unconditional.

Reply:
what do you mean by test controller? I don't get exactly where the race could happen

Reply:
I thought error injection works via some REST API from the test harness. What if the REST API happens after we reach this point?

Reply:
It looks like error injections works both via configuration and REST API. But I don't know whether it's required to make injections work on both methods.
@piodul Do you know?

In case of suppressing features, imo it's a good idea to make it work only via initial node's  configuration/

Reply:
Yes, we support enabling error injections right after configuration is parsed: https://github.com/scylladb/scylladb/pull/13861

I agree that it doesn't make sense to suppress or un-suppress cluster features in runtime.

I'm not sure though that we should go to great lengths to make impossible to change the status of the error injection in runtime - we should check its state only once on startup, as you already do, and changing it in runtime could just do nothing. This is an internal developer API, it doesn't have to be perfect.